cpp/src/arrow/compute/api_vector.cc

```diff
@@ -157,6 +157,11 @@ static auto kPartitionNthOptionsType = GetFunctionOptionsType<PartitionNthOption
 static auto kSelectKOptionsType = GetFunctionOptionsType<SelectKOptions>(
     DataMember("k", &SelectKOptions::k),
     DataMember("sort_keys", &SelectKOptions::sort_keys));
+static auto kFetchOptionsType = GetFunctionOptionsType<FetchOptions>(
```

Comment:
You can remove this whole block once fetch options is no longer a child of `FunctionOptions`.